File: php/src/CodingInterview/Domain/UserId.php
<?php

declare(strict_types=1);

namespace Folio\CodingInterview\Domain;

final class UserId
{
    public function __construct(public readonly string $value)
    {
        if ($value === '') {
            throw new \InvalidArgumentException('userId must not be empty');
        }
    }
}

File: java8/src/main/java/folio/codinginterview/domain/UserId.java
package folio.codinginterview.domain;

import java.util.Objects;

public final class UserId {
    private final String value;

    public UserId(String value) {
        if (value == null || value.isEmpty()) {
            throw new IllegalArgumentException("userId must not be empty");
        }
        this.value = value;
    }

    public String value() {
        return value;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof UserId)) return false;
        UserId other = (UserId) o;
        return Objects.equals(value, other.value);
    }

    @Override
    public int hashCode() {
        return Objects.hash(value);
    }

    @Override
    public String toString() {
        return "UserId[value=" + value + "]";
    }
}
